File: 1-two-sum/1-two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    let map = new Map();
    let result = []
    for(let i = 0; i < nums.length; i++){
        let check = target - nums[i]
        if(map.has(check)){
            result[0] = map.get(check)
            result[1] = i
        }
        map.set(nums[i], i)
    }
    return result
};

File: 1-two-sum/1-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> hm = new HashMap<>();
        int[] result = new int[2];
        for(int i = 0; i < nums.length; i++){
            int check = target - nums[i];
            if(hm.containsKey(check)){
                result[0] = i;
                result[1] = hm.get(check);
                break;
            }
            hm.put(nums[i], i);
        }
        return result;
    }
}